src/components/main-menu/MainMenu.tsx

```diff
@@ -29,41 +34,49 @@ const MainMenu = Object.assign(
       const device = useDevice();
       const appState = useUIAppState();
       const setAppState = useExcalidrawSetAppState();
+      const excaliAppState = useExcalidrawAppState();
       const onClickOutside = device.isMobile
         ? undefined
         : () => setAppState({ openMenu: null });
 
       return (
         <MainMenuTunnel.In>
-          <DropdownMenu open={appState.openMenu === "canvas"}>
-            <DropdownMenu.Trigger
-              onToggle={() => {
-                setAppState({
-                  openMenu: appState.openMenu === "canvas" ? null : "canvas",
-                });
-              }}
-              data-testid="main-menu-trigger"
-            >
-              {HamburgerMenuIcon}
-            </DropdownMenu.Trigger>
-            <DropdownMenu.Content
-              onClickOutside={onClickOutside}
-              onSelect={composeEventHandlers(onSelect, () => {
-                setAppState({ openMenu: null });
-              })}
-            >
-              {children}
-              {device.isMobile && appState.collaborators.size > 0 && (
-                <fieldset className="UserList-Wrapper">
-                  <legend>{t("labels.collaborators")}</legend>
-                  <UserList
-                    mobile={true}
-                    collaborators={appState.collaborators}
-                  />
-                </fieldset>
-              )}
-            </DropdownMenu.Content>
-          </DropdownMenu>
+          <div className="fileNameFlexBox">
+            <DropdownMenu open={appState.openMenu === "canvas"}>
+              <DropdownMenu.Trigger
+                onToggle={() => {
+                  setAppState({
+                    openMenu: appState.openMenu === "canvas" ? null : "canvas",
+                  });
+                }}
+                data-testid="main-menu-trigger"
+              >
+                {HamburgerMenuIcon}
+              </DropdownMenu.Trigger>
+              <DropdownMenu.Content
+                onClickOutside={onClickOutside}
+                onSelect={composeEventHandlers(onSelect, () => {
+                  setAppState({ openMenu: null });
+                })}
+              >
+                {children}
+                {device.isMobile && appState.collaborators.size > 0 && (
+                  <fieldset className="UserList-Wrapper">
+                    <legend>{t("labels.collaborators")}</legend>
+                    <UserList
+                      mobile={true}
+                      collaborators={appState.collaborators}
+                    />
+                  </fieldset>
+                )}
+              </DropdownMenu.Content>
+            </DropdownMenu>
+            {excaliAppState.fileHandle !== null && !device.isMobile ? (
+              <span>
+                <FileName name={excaliAppState.fileHandle.name}></FileName>
+              </span>
+            ) : null}
```

Comment:
let's render this inside here instead, as it shouldn't be part of MainMenu component:

https://github.com/excalidraw/excalidraw/blob/b57b3b573da874f7527589c10632ec936c1962d2/src/components/LayerUI.tsx#L177-L182